src/tutorial/furthertopics/message-triggers.rst

```diff
@@ -0,0 +1,439 @@
+task event handling system. More information is available on these in the 
+`Cylc User Guide`_.
+
+
+:term:`Message triggers <message trigger>` provide a superior solution to
+the problem of file system polling. We could, for example, design our suite
+such that we check if our task is finished by polling at intervals.
+It is inefficient to 'spam' task hosts with polling commands, it is preferable
+to set up a message trigger. 
+
+How to create a message trigger
+-------------------------------
+
+In order to get our suite to trigger messages, we need to:
+
+* specify our custom message in a section called ``[[outputs]]`` within the
+     ``[runtime]`` section of our suite,
+
+* add ``cylc message -- "${CYLC_SUITE_NAME}" "${CYLC_TASK_JOB}" "YOUR CHOSEN TRIGGER MESSAGE"``
+     to the ``script`` section of ``[runtime]``, your chosen trigger message 
+     should be unique and should exactly match the message defined in
+     ``[[outputs]]``.  
+
+* Refer to these messages in the ``[dependencies]`` section of our suite.
+
+These outputs are then triggered during the running of the task.    
+We can use these to manage tasks dependant on partially completed tasks.
+
+So, a basic example, where we have a task foo, that when partially completed 
+triggers another task bar and when fully completed triggers another task, baz. 
+
+   .. code-block:: cylc
+
+      [scheduling]
+         [[dependencies]]
+         graph = """foo:out1 => bar
+                     foo => baz"""
+      [runtime]
+         [[foo]]
+            script = """
+               sleep 5
+               cylc message -- "${CYLC_SUITE_NAME}" "${CYLC_TASK_JOB}" "file 1 done"
+               sleep 10
+                  """
+            [[[outputs]]]
+               out1 = "file 1 done"
+                  
+         [[bar, baz]]
+            script = sleep 10
+
+.. _message triggers practical:
+
+.. practical::
+
+   .. rubric:: In this practical we will create a suite to demonstrate
+      :term:`message triggers <message trigger>`. We will use message triggers
+      to both produce a report and trigger a new task from a partially completed
+      task.
+
+   #. **Create a new directory.**
+
+      Within your ``~/cylc-run`` directory create a new directory called 
+
+      ``message-triggers`` and move into it:
+
+      .. code-block:: bash
+
+         mkdir ~/cylc-run/message-triggers
+         cd ~/cylc-run/message-triggers
+
+   #. **Install the script needed for our suite**
+
+      The suite we will be designing requires a bash script, ``random.sh``,
+      to produce our report. It will simply create a text file ``report.txt``
+      with some random numbers in it. This will be executed when the associated
+      task is run. 
+
+      Scripts should be kept in the ``bin`` sub-directory within the 
+      :term:`suite directory <suite directory>`. If a ``/bin`` exists in the 
+      suite directory, it will be prepended to the PATH at run time.
```

Comment:
```suggestion
      suite directory, it will be prepended $PATH at run time.
```
(This may be a bit pedantic, but I'd say either $PATH, or "the executable search path").